File: 1.two-sum/solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val mem = HashMap<Int, Int>()
        nums.forEachIndexed { i, num ->
            val rest = target - num
            if (mem.containsKey(rest)) {
                return intArrayOf(i, mem.get(rest)!!)
            }
            mem.put(num, i)
        }
        return intArrayOf(0, 0)
    }
}

File: 1.two-sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
  const memo = new Map<number, number>();
  
  for (let i = 0; i < nums.length; i++) {
      const num = nums[i];
      const rest = target - num;
      if (memo.has(rest)) {
          return [memo.get(rest)!, i];
      }
      memo.set(num, i);
  }
};